subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -0,0 +1,848 @@
+		pe_send_soft_reset(dev, PD_PACKET_SOP);
+	}
+}
+
+/**
+ * @brief PE_Send_Not_Supported Exit state
+ */
+void pe_drs_send_swap_exit(void *obj)
+{
+	struct policy_engine *pe = (struct policy_engine *)obj;
+
+	/* Stop Sender Response Timer */
+	usbc_timer_stop(&pe->pd_t_sender_response);
+}
+
+static void pe_suspend_entry(void *obj)
+{
+	LOG_INF("PE_SUSPEND");
+}
+
+static void pe_suspend_run(void *obj)
+{
+	/* DO NOTHING */
+}
+
+/**
+ * @brief Policy engine State table
+ */
+static const struct smf_state pe_states[] = {
+	[PE_SNK_STARTUP] = SMF_CREATE_STATE(
+		pe_snk_startup_entry,
+		pe_snk_startup_run,
+		NULL,
+		NULL),
+	[PE_SNK_DISCOVERY] = SMF_CREATE_STATE(
+		pe_snk_discovery_entry,
+		pe_snk_discovery_run,
+		NULL,
+		NULL),
+	[PE_SNK_WAIT_FOR_CAPABILITIES] = SMF_CREATE_STATE(
+		pe_snk_wait_for_capabilities_entry,
+		pe_snk_wait_for_capabilities_run,
+		pe_snk_wait_for_capabilities_exit,
+		NULL),
+	[PE_SNK_EVALUATE_CAPABILITY] = SMF_CREATE_STATE(
+		pe_snk_evaluate_capability_entry,
+		NULL,
+		NULL,
+		NULL),
+	[PE_SNK_SELECT_CAPABILITY] = SMF_CREATE_STATE(
+		pe_snk_select_capability_entry,
+		pe_snk_select_capability_run,
+		pe_snk_select_capability_exit,
+		NULL),
+	[PE_SNK_READY] = SMF_CREATE_STATE(
+		pe_snk_ready_entry,
+		pe_snk_ready_run,
+		NULL,
+		NULL),
+	[PE_SNK_HARD_RESET] = SMF_CREATE_STATE(
+		pe_snk_hard_reset_entry,
+		pe_snk_hard_reset_run,
+		NULL,
+		NULL),
+	[PE_SNK_TRANSITION_TO_DEFAULT] = SMF_CREATE_STATE(
+		pe_snk_transition_to_default_entry,
+		pe_snk_transition_to_default_run,
+		NULL,
+		NULL),
+	[PE_SNK_GIVE_SINK_CAP] = SMF_CREATE_STATE(
+		pe_snk_give_sink_cap_entry,
+		pe_snk_give_sink_cap_run,
+		NULL,
+		NULL),
+	[PE_SNK_GET_SOURCE_CAP] = SMF_CREATE_STATE(
+		pe_snk_get_source_cap_entry,
+		pe_snk_get_source_cap_run,
+		pe_snk_get_source_cap_exit,
+		NULL),
+	[PE_SNK_TRANSITION_SINK] = SMF_CREATE_STATE(
+		pe_snk_transition_sink_entry,
+		pe_snk_transition_sink_run,
+		pe_snk_transition_sink_exit,
+		NULL),
+	[PE_SEND_SOFT_RESET] = SMF_CREATE_STATE(
+		pe_send_soft_reset_entry,
+		pe_send_soft_reset_run,
+		pe_send_soft_reset_exit,
+		NULL),
+	[PE_SOFT_RESET] = SMF_CREATE_STATE(
+		pe_soft_reset_entry,
+		pe_soft_reset_run,
+		NULL,
+		NULL),
+	[PE_SEND_NOT_SUPPORTED] = SMF_CREATE_STATE(
+		pe_send_not_supported_entry,
+		pe_send_not_supported_run,
+		NULL,
+		NULL),
+	[PE_DRS_EVALUATE_SWAP] = SMF_CREATE_STATE(
+		pe_drs_evaluate_swap_entry,
+		pe_drs_evaluate_swap_run,
+		NULL,
+		NULL),
+	[PE_DRS_SEND_SWAP] = SMF_CREATE_STATE(
+		pe_drs_send_swap_entry,
+		pe_drs_send_swap_run,
+		pe_drs_send_swap_exit,
+		NULL),
+	[PE_SNK_GET_SOURCE_CAP] = SMF_CREATE_STATE(
+		pe_snk_get_source_cap_entry,
+		pe_snk_get_source_cap_run,
+		pe_snk_get_source_cap_exit,
+		NULL),
+	[PE_SNK_CHUNK_RECEIVED] = SMF_CREATE_STATE(
+		pe_chunk_received_entry,
+		pe_chunk_received_run,
+		NULL,
+		NULL),
+	[PE_SUSPEND] = SMF_CREATE_STATE(
+		pe_suspend_entry,
+		pe_suspend_run,
+		NULL,
+		NULL),
+};
```

Comment:
This needs a build time check that `ARRAY_SIZE(pe_states) == PE_STATE_COUNT`.  Where `PE_STATE_COUNT = PE_SUSPEND+1`.

Reply:
Done